src/MultiRegion/unified_multi_region_fft_solver.jl

```diff
@@ -0,0 +1,188 @@
+using Oceananigans.Architectures: device_event
+using FFTW
+using CUDA, CUDA.CUFFT
+using Oceananigans.Operators
+
+import Oceananigans.Architectures: architecture
+import Oceananigans.Models.NonhydrostaticModels: PressureSolver, solve_for_pressure!
+import Oceananigans.Solvers: FFTBasedPoissonSolver, FourierTridiagonalPoissonSolver, solve!
+
+using Oceananigans.Grids 
+using Oceananigans.Solvers: poisson_eigenvalues, plan_transforms
+
+struct UnifiedMultiRegionPoissonSolver{G, S}
+    grid :: G
+    solver :: S
+end
+
+PressureSolver(arch, grid::ImmersedMultiRegionGrid, planner_flag=FFT.PATIENT) = PressureSolver(arch, grid.underlying_grid, planner_flag)
```

Comment:
```suggestion
PressureSolver(arch, grid::ImmersedMultiRegionGrid, planner_flag=FFTW.PATIENT) = PressureSolver(arch, grid.underlying_grid, planner_flag)
```

I'm getting a `UndefVarError: FFT not defined` error here.